apps/src/schoolInfo/hooks/useSchoolInfo.tsx

```diff
@@ -62,10 +62,10 @@ export function useSchoolInfo(initialState: SchoolInfoInitialState) {
 
   const detectedSchoolName = useMemo(
     () =>
-      initialState.schoolName ||
+      (initialState.schoolId ? '' : initialState.schoolName) ||
```

Comment:
form state should not necessarily match the initial state. the `current_school` method returns the nces school name when all the form needs is the `schoolId` and `schoolZip` to populate the form. `schoolName` form state should be reserved for user input when their school is not found in the list of nces schools, which is determined by a lack of `initialState.schoolId`